transaction.go

```diff
@@ -2,10 +2,66 @@ package seth
 
 import (
 	"bytes"
-	"io"
 )
 
-func intBytes(n Uint64) []byte {
+// Transaction represents an ethereum transaction.
+type Transaction struct {
+	Hash        Hash     `json:"hash"`             // tx hash
+	Nonce       Uint64   `json:"nonce"`            // sender nonce
+	Block       Hash     `json:"blockHash"`        // hash of parent block
+	BlockNumber Uint64   `json:"blockNumber"`      //
+	To          *Address `json:"to"`               // receiver, or nil for contract creation
+	TxIndex     *Uint64  `json:"transactionIndex"` // transaction index, or nil if pending
+	From        *Address `json:"from"`             // from
+	Value       Int      `json:"value"`            // value in wei
+	GasPrice    Int      `json:"gasPrice"`         // gas price
+	Gas         Uint64   `json:"gas"`              // gas spent on transaction
+	Input       Data     `json:"input"`            // input data
+}
+
+// Encode returns an RLP encoded representation of the transaction. If a
+// signature is provided, this will return an encoded representation containing
+// the signature.
+func (t *Transaction) Encode(sig *Signature) []byte {
+	var e rlpEncoder
+	if sig == nil {
+		e.EncodeTransaction(t)
+	} else {
+		e.EncodeSignedTx(t, sig)
+	}
+	return e.Bytes()
+}
+
+// HashToSign returns a hash which can be used to sign the transaction.
+func (t *Transaction) HashToSign() *Hash {
```

Comment:
Is this different from the transaction hash?

Reply:
Edit: it must not be. This hash doesn't cover the sender. (It's disappointing that the tx hash depends upon the nondeterministically derived signature.)